### PR TITLE
Add fixes

### PR DESCRIPTION
**Pull request by Atry**
The fixes are required to build a poetry project imported from https://github.com/oobabooga/text-generation-webui/blob/main/requirements.txt

### CODE REVIEW

File: overrides/default.nix [hunk 1]
```diff
@@ -239,6 +239,12 @@ lib.composeManyExtensions [
         }
       );
 
+      bitsandbytes = super.bitsandbytes.overridePythonAttrs (
+        old: {
+          propagatedBuildInputs = (old.propagatedBuildInputs or [ ]) ++ [ self.setuptools ];
```
Comment (cpcloud) [inline overrides/default.nix:244]: Can you move all of these `setuptools` additions to `build-systems.json`?

File: overrides/default.nix [hunk 2]
```diff
@@ -1591,7 +1632,7 @@ lib.composeManyExtensions [
       pandas = super.pandas.overridePythonAttrs (old: {
 
         buildInputs = old.buildInputs or [ ] ++ lib.optional stdenv.isDarwin pkgs.libcxx;
-
+        propagatedBuildInputs = (old.propagatedBuildInputs or [ ]) ++ [ self.versioneer ];
```
Comment (cpcloud) [inline overrides/default.nix:1635]: This should also go in `build-systems.json`

File: overrides/default.nix [hunk 3]
```diff
@@ -1086,6 +1111,22 @@ lib.composeManyExtensions [
         }
       );
 
+      llama-cpp-python = super.llama-cpp-python.overridePythonAttrs (
+        old: {
+          propagatedBuildInputs = (old.propagatedBuildInputs or [ ]) ++ [
+            self.setuptools
+            self.scikit-build
```
Comment (cpcloud) [inline overrides/default.nix:1118]: Any Python library that's needed to build or install the library, but not to run it, should go in `build-systems.json`.

File: overrides/default.nix [hunk 4]
```diff
@@ -305,6 +311,19 @@ lib.composeManyExtensions [
...
+      #     cmakeFlags = [
+      #       "-DCMAKE_OSX_DEPLOYMENT_TARGET=10.10"
+      #     ];
+      #     preBuild = ''
+      #       cd "$OLDPWD"
+      #     '';
+      #     nativeBuildInputs = old.nativeBuildInputs or [ ] ++ [ pkgs.cmake pkg-config ];
+      #     propagatedBuildInputs = (old.propagatedBuildInputs or [ ]) ++ [ self.setuptools self.scikit-build ];
+      #   }
+      # );
```
Comment (cpcloud) [inline overrides/default.nix:325]: Remove this if it isn't being used.

File: overrides/default.nix [hunk 5]
```diff
@@ -1651,6 +1692,12 @@ lib.composeManyExtensions [
         }
       );
 
+      peft = super.peft.overridePythonAttrs (
+        old: {
+          propagatedBuildInputs = (old.propagatedBuildInputs or [ ]) ++ [ self.setuptools ];
```
Comment (cpcloud) [inline overrides/default.nix:1697]: Needs to go in `build-systems.json`

File: overrides/default.nix [hunk 6]
```diff
@@ -1799,6 +1846,10 @@ lib.composeManyExtensions [
                   pkgs.cmake
                 ];
 
+                buildInputs = (old.buildInputs or [ ]) ++ [
+                  _arrow-cpp
```
Comment (cpcloud) [inline overrides/default.nix:1850]: Are you sure this is necessary?

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: overrides/default.nix (hunk 6 above)
Comment (Atry) [reply overrides/default.nix:1831]: Yes, at least on macOS

**Comment by Atry** [comment]
Most of the fixes have been added by other people in other PRs, so I simply removed them.

**Review by cpcloud — APPROVED** [review]

**Pull request merged**